File: 05-non-constructible-change.py
def non_constructible_change(coins):
    coins.sort()

    current_change_created = 0

    for coin in coins:
        if coin > current_change_created + 1:
            return current_change_created + 1

        current_change_created += coin

    return current_change_created + 1

File: 05-non-constructible-change.js
// Time - O(nlog(n))
// Space - O(1)

function nonConstructibleChange(coins) {
  // sort coins
  coins.sort((a, b) => a - b);

  let currentChangeCreated = 0;
  for (const coin of coins) {
    if (coin > currentChangeCreated + 1) return currentChangeCreated + 1;

    currentChangeCreated += coin;
  }

  return currentChangeCreated + 1;
}
